decidim/decidim
# Generate component Gemfile template when releasing

#### :tophat: What? Why?
This PR tries to fix the issue described in https://github.com/decidim/decidim/issues/11754. The idea of this fix is to generate the Gemfile.erb template at the release time to ensure consistent version between dependencies. Even though just faker is impacted, I have noticed there are some other gems that could be updated.

Note to reviewer: 
- It seemed more simple to override the entire file, than to manually patch line by line ( with gsub etc )
- I thought to add a note in generated Erb file, but i quit that idea, as the comment will end up in the generated module. 

#### :pushpin: Related Issues
*Link your PR to an issue*
- Fixes #11754

#### Testing
1. Observe `decidim-generators/lib/decidim/generators/component_templates/Gemfile.erb` contents
2. Empty the file
3. Run `bin/rake patch_generators`
4. Observe the new content

:hearts: Thank you!
